nixos/modules/services/networking/mosh.nix

```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg  = config.services.mosh;
```

Comment:
I did think about doing that, but I decided to put it in `services` because this module opens up ports in the firewall. I'd expect that from a service, but I'd be very surprised if a  `program` did that. Another reason is that I think of mosh as essentially 'better ssh', and this is configuring the mosh 'server' end, not the 'client' end.

Mosh is definitely a bit weird in this, so I'm ok with changing this if you feel otherwise.